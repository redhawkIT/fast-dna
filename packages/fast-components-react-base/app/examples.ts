import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };
