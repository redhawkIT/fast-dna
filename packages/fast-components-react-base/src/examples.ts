import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: Array<IExampleData<T>>;
}
