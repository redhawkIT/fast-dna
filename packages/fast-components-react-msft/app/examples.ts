import ButtonExample from "../src/button/examples.data";
export {ButtonExample};

import DividerExample from "../src/divider/examples.data";
export {DividerExample};

import ToggleExamples from "../src/toggle/examples.data";
export {ToggleExamples};
