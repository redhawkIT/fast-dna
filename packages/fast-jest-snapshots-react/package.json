{
  "name": "@microsoft/fast-jest-snapshots-react",
  "description": "A small utility for automating Jest snapshot creation of React components",
  "version": "1.6.0",
  "author": {
    "name": "Microsoft",
    "email": "fwq@microsoft.com"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Microsoft/fast-dna.git"
  },
  "private": true,
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "prepare": "tsc -p ./tsconfig.json",
    "test": "npm run tslint",
    "start": "webpack-dev-server",
    "tslint": "tslint -c ./tslint.json '**/*.ts{,x}'",
    "tslint:fix": "tslint -c ./tslint.json --fix '**/*.ts{,x}'"
  },
  "devDependencies": {
    "@microsoft/fast-tslint-rules": "^1.6.0",
    "@types/jest": "^22.2.2",
    "jest": "^22.4.3",
    "react": "^16.3.0",
    "ts-jest": "^22.4.2",
    "tslint": "^5.9.1",
    "typescript": "^2.8.1"
  },
  "dependencies": {
    "@types/react": "^16.3.0",
    "react-test-renderer": "^16.3.0"
  },
  "peerDependencies": {
    "jest": "^22.0.0",
    "react": "^16.3.0"
  }
}
