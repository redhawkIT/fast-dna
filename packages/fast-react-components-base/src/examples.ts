import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: IExampleData<T>[];
}