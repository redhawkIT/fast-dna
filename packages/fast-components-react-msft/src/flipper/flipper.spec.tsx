import * as React from "react";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import MSFTFlipper, {
    FlipperProps,
    IFlipperHandledProps,
    IFlipperManagedClasses,
    IFlipperUnhandledProps
} from "./flipper";
import { Flipper } from "./index";

/*
 * Configure Enzyme
 */
configure({adapter: new Adapter()});

describe("flipper snapshots", (): void => {
    generateSnapshots(examples);
});

describe("flipper", (): void => {
    test("should have a displayName that matches the component name", () => {
        expect((MSFTFlipper as any).name).toBe(MSFTFlipper.displayName);
    });

    test("should not throw if managedClasses are not provided", () => {
        expect(
            () => {
                shallow(<MSFTFlipper />);
            }
        ).not.toThrow();
    });

    test("should return an object that includes all valid props which are not enumerated as handledProps", () => {
        const handledProps: IFlipperHandledProps = {
            visibleToAssistiveTechnologies: false
        };

        const unhandledProps: IFlipperUnhandledProps = {
            "aria-labelledby": "foo"
        };

        const props: IFlipperHandledProps & IFlipperUnhandledProps = {...handledProps, ...unhandledProps};

        const rendered: any = shallow(
            <Flipper {...props} />
        );

        const flipper: any = rendered.first().shallow();

        expect(flipper.prop("aria-labelledby")).toEqual("foo");
    });

    test("should set an attribute of `tabindex` to -1 if `visibility` prop is false", () => {
        const props: IFlipperHandledProps = {
            visibleToAssistiveTechnologies: false
        };

        const rendered: any = shallow(
            <Flipper {...props} />
        );

        const flipper: any = rendered.first().shallow();

        expect(flipper.prop("tabIndex")).toEqual(-1);
    });

    test("should set an attribute of `aria-hidden` to true if `visibility` prop is false", () => {
        const props: IFlipperHandledProps = {
            visibleToAssistiveTechnologies: false
        };

        const rendered: any = shallow(
            <Flipper {...props} />
        );

        const flipper: any = rendered.first().shallow();

        expect(flipper.prop("aria-hidden")).toEqual(true);
    });

    test("should not set an attribute of `aria-label` if no label is passed", () => {
        const props: IFlipperHandledProps = {
            visibleToAssistiveTechnologies: false
        };

        const rendered: any = shallow(
            <Flipper {...props} />
        );

        const flipper: any = rendered.first().shallow();

        expect(flipper.prop("aria-label")).toBe(undefined);
    });

    test("should set an attribute of `aria-label` if `label` prop is passed", () => {
        const props: IFlipperHandledProps = {
            visibleToAssistiveTechnologies: true,
            label: "Test aria-label"
        };

        const rendered: any = shallow(
            <Flipper {...props} />
        );

        const flipper: any = rendered.first().shallow();

        expect(flipper.instance().props.label).toBe("Test aria-label");
        expect(flipper.prop("aria-label")).toBe("Test aria-label");
    });
});
