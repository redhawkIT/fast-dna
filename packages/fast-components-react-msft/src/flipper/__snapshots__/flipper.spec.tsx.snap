// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flipper snapshots Flipper: 0 1`] = `
<button
  aria-hidden={true}
  className="button-0-1-5 flipper_next-0-1-7"
  data-sketch-symbol="Flipper - default"
  href={null}
  tabIndex={-1}
>
  <span
    className="flipper_glyph-0-1-6"
  />
</button>
`;

exports[`flipper snapshots Flipper: 1 1`] = `
<button
  aria-hidden={true}
  className="button-0-1-13 flipper_previous-0-1-16"
  direction="previous"
  href={null}
  tabIndex={-1}
>
  <span
    className="flipper_glyph-0-1-14"
  />
</button>
`;

exports[`flipper snapshots Flipper: 2 1`] = `
<button
  aria-label="See next"
  className="button-0-1-21 flipper_next-0-1-23"
  direction="next"
  href={null}
>
  <span
    className="flipper_glyph-0-1-22"
  />
</button>
`;

exports[`flipper snapshots Flipper: 3 1`] = `
<button
  aria-label="See previous"
  className="button-0-1-29 flipper_previous-0-1-32"
  direction="previous"
  href={null}
>
  <span
    className="flipper_glyph-0-1-30"
  />
</button>
`;
